File: org.eclipse.osee.framework.ui.plugin/src/org/eclipse/osee/framework/ui/plugin/util/Files.java
/*******************************************************************************
 * Copyright (c) 2004, 2007 Boeing.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Boeing - initial API and implementation
 *******************************************************************************/
package org.eclipse.osee.framework.ui.plugin.util;

import java.io.File;
import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.FileDialog;
import org.eclipse.swt.widgets.Shell;

/**
 * @author Robert A. Fisher
 */
public class Files {

   public static File selectFile(Shell shell, int swtType, String... extensions) {
      FileDialog dialog = new FileDialog(shell, swtType | SWT.SINGLE);
      dialog.setFilterExtensions(extensions);
      dialog.setFilterPath(AWorkspace.getWorkspacePath());

      String path = dialog.open();

      if (path != null) {
         return new File(path);
      } else {
         return null;
      }
   }
}

File: web/apps/osee/src/app/shared/types/files.ts
/*********************************************************************
 * Copyright (c) 2022 Boeing
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Boeing - initial API and implementation
 **********************************************************************/
export const enum FileExtensions {
	XML = 'xml',
	ZIP = 'zip',
	CSV = 'csv',
	JSON = 'json',
}

export const enum ProducesMediaType {
	JSON = 'application/json',
	XML = 'application/xml',
	ZIP = 'application/zip',
}

// (mebibytes) by (bytes per kibibyte) by (kibibytes per mebibyte)
export const MAX_FILE_SIZE_BYTES = 50 * 1024 * 1024;
